### DISCUSSION TITLE
Does anyone know why the last API was added to libcrypto.num with an ID of "?"

### DISCUSSION THREAD

**Discussion by slontis** [openssl/openssl#22979] [Q&A] [answered]
https://github.com/openssl/openssl/blob/master/util/libcrypto.num
EVP_DigestSqueeze                       ?	3_3_0	EXIST::FUNCTION:

**Comment by levitte** [comment, marked as answer]
Yes.  Its ordinal number hasn't been determined yet.  The numbers get added when we make a release.

This was done to avoid merge conflicts and having to renumber them when work is being done in multiple branches (i.e. PRs) which may be merged back into the base branch in any order.
Before we came up with this scheme, having to renumber the last bits was no fun exercise, let me tell ya!

**Reply by mspncp** [reply]
To be precise: merge conflicts still happen, because all new symbols get inserted at the end. But now the conflicts can be resolved trivially by rerunning make update without caring about renumbering the new symbols, right @levitte?

The other half of the plan was to sort the symbols alphabetically, not numerically, to spread out the inserts and reduce merge conflicts, but that didn't happen.
